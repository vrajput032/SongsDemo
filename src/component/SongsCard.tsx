import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import { vw, fullWidth, vh } from '../service/dimension';
import Image from './Image';
import { parser } from '../service/parser';

interface Props {
  rowData: { item: any; index: number };
  pushToDetails: (item: any) => void;
}

const SongsCard = React.memo(function Card(props: Props) {
  let name = props.rowData.item?.trackName?.toUpperCase()
  return (
    <TouchableOpacity
      onPress={() => props.pushToDetails(props.rowData.item)}
      style={styles.container}>
      <Image
        source={{
          uri: parser(props.rowData.item.artworkUrl100, 'source/300x300'),
        }}
        style={styles.imageStyle}
        data-test="Album-card-component"
      />
      <Text
        style={styles.title}
        numberOfLines={1}>
        {`${name}`}</Text>
    </TouchableOpacity>
  );
});

export default SongsCard;

const styles = StyleSheet.create({
  container: {
    width: fullWidth / 2 - 30,
    marginHorizontal: 5,
    marginVertical: 15,
    borderRadius: 5
  },
  imageStyle: {
    borderRadius: 15,
    height: vw(150),
    width: vw(150),
  },
  title: {
    marginTop: 15,
    color:'white',
    fontSize: vw(16),
    alignSelf:'center',
    fontWeight: 'bold',
  },
});
