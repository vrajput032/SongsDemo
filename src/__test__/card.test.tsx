/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import { shallow } from 'enzyme';
import SongsCard from '../component/SongsCard';
const findByTestAttr = (component: any, attr: any) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
describe('Songs Card Component', () => {
  const wrapperCardComponent = shallow(
    <SongsCard
      rowData={{ item: {}, index: 0 }}
      pushToDetails={(_item: any) => {}}
    />,
  );
  it('Songs snapshot', () => {
    expect(wrapperCardComponent).toMatchSnapshot('GP Card snap');
  });
  it('Songs card render', () => {
    const component = findByTestAttr(
      wrapperCardComponent,
      'Songs-card-component',
    );
    expect(component.length).toBe(0);
  });
});
