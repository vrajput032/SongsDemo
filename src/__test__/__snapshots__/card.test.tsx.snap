// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs Card Component Songs snapshot: GP Card snap 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "borderRadius": 5,
      "marginHorizontal": 5,
      "marginVertical": 15,
      "width": 345,
    }
  }
>
  <Image
    data-test="Album-card-component"
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 15,
        "height": 300,
        "width": 300,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 32,
        "fontWeight": "bold",
        "marginTop": 15,
      }
    }
  >
    undefined
  </Text>
</ForwardRef>
`;
